compiler/resolution/generics.cpp

```diff
@@ -179,24 +198,39 @@ checkInfiniteWhereInstantiation(FnSymbol* fn) {
 //
 static void
 checkInstantiationLimit(FnSymbol* fn) {
-  static Map<FnSymbol*,int> instantiationLimitMap;
-
-  // Don't count instantiations on internal modules
-  // nor ones explicitly marked NO_INSTANTIATION_LIMIT.
-  if (fn->getModule() &&
-      fn->getModule()->modTag != MOD_INTERNAL &&
-      !fn->hasFlag(FLAG_NO_INSTANTIATION_LIMIT)) {
+  if (trackInstantiationsForFn(fn)) {
     if (instantiationLimitMap.get(fn) >= instantiation_limit) {
       USR_FATAL_CONT(fn, "Function '%s' has been instantiated too many times",
                      fn->name);
       USR_PRINT("  If this is intentional, try increasing"
                 " the instantiation limit from %d", instantiation_limit);
       USR_STOP();
     }
+    /*
+    printf("Incrementing instantiation count for %s (%p)\n",
+           fn->name, fn);
+    */
```

Comment:
I am not excited about having this instrumentation on master.

Reply:
I'd be OK if it fit all in a single line.
Ditto the one for popping.

Reply:
You're saying you're OK with 1-line commented out printf()s but not 2-line commented out printf()s?

Reply:
Correct, under the condition that the comment markers do not occupy any additional lines.